pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -258,3 +266,21 @@ func (dv *openShiftClusterDynamicValidator) validateSubnet(ctx context.Context,
 
 	return net, nil
 }
+
+// validateVnet checks that the vnet does not have custom dns servers set
+func (dv *openShiftClusterDynamicValidator) validateVnet(ctx context.Context, m utilvnet.Manager, oc *api.OpenShiftCluster) error {
+	vnetID, _, err := subnet.Split(oc.Properties.MasterProfile.SubnetID)
+	if err != nil {
+		return err
+	}
+	vnet, err := m.Get(ctx, vnetID)
+	if err != nil {
+		return err
+	}
+	servers := to.StringSlice(vnet.DhcpOptions.DNSServers)
+	if len(servers) > 0 {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidLinkedVNet, "properties.masterProfile.subnetId", "The provided vnetID  '%s' is invalid: custom DNS servers are not supported.", vnetID)
```

Comment:
I'm parsing the subnet above to fetch the vnetID.  I wasn't sure where else to find the vnet id.

When vnet custom dns is set it returns a list of strings.  When it is not set, the list is length 0.  Is this sufficient for the check?

Reply:
all good.